src/app/qgsdockablewidget.cpp

```diff
@@ -0,0 +1,119 @@
+/***************************************************************************
+  qgsdockablewidget.cpp
+  --------------------------------------
+  Date                 : July 2017
+  Copyright            : (C) 2017 by Martin Dobias
+  Email                : wonder dot sk at gmail dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "qgsdockablewidget.h"
+
+#include "qgisapp.h"
+#include "qgs3dmapcanvaswidget.h"
+#include "qgsdockwidget.h"
+
+#include <QWidget>
+
+QgsDockableWidget::QgsDockableWidget( QWidget *parent )
+  : QWidget( parent )
+{
+  this->setAttribute( Qt::WA_DeleteOnClose );
```

Comment:
It is unclear to me why we're using DeleteOnClose here and on some other objects...

Reply:
It is because elsewhere we want to just call close and let the widget delete itself later, same for the QDialog and QgsDockWidget. Honestly I do not see how to handle making the dockable widget fire a closed signal and then be deleted in other parts of the code without having Qt::WA_DeleteOnClose.